torch/csrc/distributed/c10d/init.cpp

```diff
@@ -3161,9 +3161,52 @@ such as `dist.all_reduce(tensor, async_op=True)`.
         Arguments:
           tensors(List[torch.Tensor]): List of tensors we want to hash.
       )");
-  module.def("_dump_nccl_trace", []() {
-    return py::bytes(::c10d::dump_nccl_trace());
-  });
+  module.def(
+      "_get_collective_trace",
+      [](std::optional<bool> includeStackTraces,
+         std::optional<bool> onlyActive) {
+        return py::bytes(::c10d::get_collective_trace(
+            includeStackTraces.value_or(true), onlyActive.value_or(false)));
+      },
+      py::arg("includeStackTraces") = std::optional<bool>(),
+      py::arg("onlyActive") = std::optional<bool>(),
+      R"(
+        Arguments:
+            includeStackTraces(bool, optional): Whether to include stacktraces in the collective work traces. Default is True.
+            onlyActive (bool, optional): Whether to only include active collective work traces. Default is False.
+        Returns:
+            Stringified pickle collective work traces.
+            Default settings return all collectives with stack traces.
+      )");
+  module.def(
+      "_get_nccl_comm_trace",
```

Comment:
Similarly, this API helps get at the nccl_comm_trace but not get everything else.

Main idea is that the `get_` API favor retrieval over HTTP